forc-tracing/src/lib.rs

```
@@ -265,44 +305,116 @@ pub fn init_tracing_subscriber(options: TracingSubscriberOptions) {
+    let regex_filter = options.regex_filter.clone();
+    let regex_filter_fn = filter_fn(move |metadata| {
+        if let Some(ref regex_filter) = regex_filter {
             let regex = regex::Regex::new(regex_filter).unwrap();
             regex.is_match(metadata.target())
         } else {
             true
         }
     });
 
-    match (is_json_mode_active(), level_filter) {
-        (true, Some(level)) => {
-            let subscriber = builder.json().with_max_level(level).finish().with(filter);
-            tracing::subscriber::set_global_default(subscriber).expect("setting subscriber failed");
-        }
-        (true, None) => {
-            let subscriber = builder.json().finish().with(filter);
-            tracing::subscriber::set_global_default(subscriber).expect("setting subscriber failed");
-        }
-        (false, Some(level)) => {
-            let subscriber = builder.with_max_level(level).finish().with(filter);
-            tracing::subscriber::set_global_default(subscriber).expect("setting subscriber failed");
-        }
-        (false, None) => {
-            let subscriber = builder.finish().with(filter);
-            tracing::subscriber::set_global_default(subscriber).expect("setting subscriber failed");
+    // Create filters for non-telemetry mode
+    let composite_filter = env_filter.and(hide_telemetry_filter).and(regex_filter_fn);
+
+    // Initialize registry with explicit layer handling
+    #[cfg(feature = "telemetry")]
+    {
+        if !disabled {
+            if let Ok((telemetry_layer, guard)) = fuel_telemetry::new_with_watchers!() {
+                // Store the WorkerGuard in thread-local storage
+                TELEMETRY_GUARD.with(|g| {
+                    *g.borrow_mut() = Some(guard);
+                });
+
+                // Use the HideTelemetryFilter to prevent telemetry spans from appearing in fmt output
+                let hide_filter = HideTelemetryFilter;
```

Seems like fmt layer only applies `hide_filter` and `level_filter`; not `env_filter` (from `RUST_LOG`) or `regex_filter_fn`.
This means environment-based filtering and regex filtering are silently ignored when telemetry is active.

Do we want to use:
```rust
.with_filter(composite_filter)
```
instead of `.with_filter(hide_filter)` in both conditions below?

---

Oo yes, good one -- addressed in [3772f9e](https://github.com/FuelLabs/sway/pull/7295/commits/3772f9e28f43058864d33a619c1634796a85a39d)